internal/cmd/hana/map.go

```diff
@@ -66,6 +66,13 @@ func runMap(config *hanaCheckConfig) error {
 }
 
 func createHanaAPIInstance(config *hanaCheckConfig) error {
+	//check if instance exists, skip API creation if it does
+	instance, err := getServiceInstance(config, hanaBindingAPIName(config.name))
```

Comment:
Could we extract this code to another function? This check is not main responsibility of `createHanaAPIInstance`.